### PR TITLE
CI: Upload wheels to scipp pypi index

### PR DESCRIPTION
**Pull request by MridulS**
We can push wheels to pypi.anaconda.org/scipp and then we can use something like:
```
python -m pip install \
  --pre \
  --index-url https://pypi.anaconda.org/scipp/simple \
  --extra-index-url https://pypi.org/simple \
  scipp
```
to get nightly wheels instead of renaming built wheels and creating static urls.

We also should have something like https://github.com/scientific-python/upload-nightly-action/blob/main/.github/workflows/remove-wheels.yml to clean up old wheels, but let's test this infrastructure out first. If we are happy, we can further iterate on this :)

### CODE REVIEW

**Comment by jl-wynen** [comment]
In your example command, why is anaconda the `index-url` and PyPI `extra-index-url`? Does the order matter?

**Comment by jl-wynen** [comment]
Please explain this setup in the dev docs. E.g., https://scipp.github.io/development/releasing-scipp.html

**Comment by MridulS** [comment]
The order does matter, with pip `index-url` is the main index and `extra-index-url` is the fallback one. But if we start using uv it actually switches this order and breaks away from pip compatibilty https://github.com/astral-sh/uv/pull/2083, so with uv this should be:
```
python -m uv pip install \
  --pre \
  --extra-index-url https://pypi.anaconda.org/scipp/simple \
  scipp
```

**Comment by MridulS** [comment]
The triggered did upload all the wheels to https://pypi.anaconda.org/scipp/simple/scipp !
But it is also uploading these random doc links !? I'm not sure where it's getting these things.

**Comment by jl-wynen** [comment]
The links all point to the latest docs. I don't think they are a big issue if they simply get overwritten. 

Does the action overwrite existing wheels?

**Comment by MridulS** [comment]
> Does the action overwrite existing wheels?

No, it will only overwrite if they had the same name but they won't as it uses the commit hash in the name. We would need to setup a cleaning action to make sure we don't keep the old wheels around. https://github.com/scientific-python/upload-nightly-action/blob/main/.github/workflows/remove-wheels.yml as an example.

**Comment by jl-wynen** [comment]
This looks pretty complicated. Do we really need all of this? Or is there a simple way to wipe the index every night before upload?

File: .github/workflows/release-nightly.yml
Comment (jl-wynen) [inline .github/workflows/release-nightly.yml:47]: This looks like it lists both conda packages and wheels. Is this correct? Can we go via the PyPI index instead (https://pypi.anaconda.org/scipp/simple/scipp/)?

Comment (MridulS) [reply .github/workflows/release-nightly.yml:47]: I couldn't find an api end point just for pypi, the `contains("dev")` is to fetch only versions with `dev` in it's name, only the wheels have that in the name.

Comment (jl-wynen) [reply .github/workflows/release-nightly.yml:46]: > only the wheels have that in the name.

Sure, currently. It's probably fine. I just want to be extra careful when erasing data so we don't accidentally erase anything that is supposed to be permanent.

Comment (MridulS) [reply .github/workflows/release-nightly.yml:46]: We can use a new org like https://anaconda.org/scipp-nightly-wheels/ to be safe.

Comment (jl-wynen) [reply .github/workflows/release-nightly.yml:39]: I'd be more comfortable with this. But I'd like input from @nvaytet and @SimonHeybrock.

Comment (nvaytet) [reply .github/workflows/release-nightly.yml:39]: :+1: from me for `scipp-nightly-wheels` organization

Comment (MridulS) [reply .github/workflows/release-nightly.yml:42]: We are just waiting for +1 from @SimonHeybrock right?

File: .github/workflows/release-nightly.yml
Comment (MridulS) [reply .github/workflows/release-nightly.yml:4]: yes, just for testing in this PR.

**Comment by MridulS** [comment]
I'm waiting for a new release upstream https://github.com/scientific-python/upload-nightly-action first before merging this.

File: .github/workflows/release-nightly.yml
Comment (matthewfeickert) [inline .github/workflows/release-nightly.yml:45]:
```suggestion
        uses: scientific-python/upload-nightly-action@82396a2ed4269ba06c6b2988bb4fd568ef3c3d6b # 0.6.1
```

Thanks for your work!

File: .github/workflows/release-nightly.yml
Comment (matthewfeickert) [inline .github/workflows/release-nightly.yml:55]: While the `pixi` setup from the `prefix-dev/setup-pixi` should still be around, it might be safer to try to set it up again as this could potentially get cleaned up in the future.

Nice use of `pixi exec --spec` though! :)

Comment (MridulS) [reply .github/workflows/release-nightly.yml:55]: Yeah that's probably better, but we may cycle through the removing-wheels-from-anaconda bit later again :)

File: .github/workflows/release-nightly.yml
Comment (nvaytet) [inline .github/workflows/release-nightly.yml:24]: I would keep all versions of ubuntu the same across all CI? I think it's still 22.04 in other places.
That said, it's probably time to bump that version everywhere?

File: .github/workflows/release-nightly.yml
Comment (nvaytet) [inline .github/workflows/release-nightly.yml:22]: Should we make this into 2 separate jobs? One to release on github, the other on pypi?
Would make it clearer what steps belong to what, and also easier to remove the github upload in the future if we want to do that?
I guess both jobs could run in parallel?

File: .github/workflows/release-nightly.yml
Comment (nvaytet) [inline .github/workflows/release-nightly.yml:55]: I don't know if we need to checkout in this step? I think above it was needed because we run the `rename-dev-wheels.py` script?

File: .github/workflows/release-nightly.yml
Comment (nvaytet) [inline .github/workflows/release-nightly.yml:53]: Is this permissions setting also needed here or is it only relevant for the step above (and was accidentally copy/pasted)?

**Review by nvaytet — APPROVED** [review]

**Pull request merged**
